src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala

```diff
@@ -0,0 +1,126 @@
+import io.iohk.ethereum.rpc.{JsonRpcServer, RpcServerConfig}
+import io.iohk.ethereum.utils.{BlockchainStatus, Config, NodeStatus, ServerStatus}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import io.iohk.ethereum.network._
+
+import scala.concurrent.duration.Duration
+
+trait NodeKeyBuilder {
+  lazy val nodeKey = loadAsymmetricCipherKeyPair(Config.keysFile)
+}
+
+trait ActorSystemBuilder {
+  implicit lazy val actorSystem = ActorSystem("etc-client_system")
+}
+
+trait StorageBuilder {
+  lazy val storagesInstance =  new SharedLevelDBDataSources with Storages.DefaultStorages
+}
+
+trait NodeStatusBuilder {
+
+  self : NodeKeyBuilder =>
+
+  private lazy val nodeStatus =
+    NodeStatus(
+      key = nodeKey,
+      serverStatus = ServerStatus.NotListening,
+      blockchainStatus = BlockchainStatus(Config.Blockchain.genesisDifficulty, Config.Blockchain.genesisHash, 0))
+
+  lazy val nodeStatusHolder = Agent(nodeStatus)
+}
+
+trait BlockChainBuilder {
+  self: StorageBuilder =>
+
+  lazy val blockchain: Blockchain = BlockchainImpl(storagesInstance.storages)
+}
+
+
+trait PeerManagerActorBuilder {
+
+  self: ActorSystemBuilder
+    with NodeStatusBuilder
+    with BlockChainBuilder =>
+
+  lazy val peerConfiguration = Config.Network.peer
+
+  lazy val peerManager = actorSystem.actorOf(
+    PeerManagerActor.props(nodeStatusHolder, peerConfiguration, blockchain), "peer-manager")
+}
+
+trait ServerActorBuilder {
+
+  self: ActorSystemBuilder
+    with NodeStatusBuilder
+    with BlockChainBuilder
+    with PeerManagerActorBuilder =>
+
+  lazy val networkConfig = Config.Network
+
+  lazy val server = actorSystem.actorOf(ServerActor.props(nodeStatusHolder, peerManager), "server")
+
+}
+
+
+trait JSONRpcServerBuilder {
+
+  self: ActorSystemBuilder with BlockChainBuilder =>
+
+  def startJSONRpcServer(): Unit = JsonRpcServer.run(actorSystem, blockchain, Config.Network.Rpc)
+
+  lazy val rpcServerConfig: RpcServerConfig = Config.Network.Rpc
+
+}
+
+trait FastSyncControllerBuilder {
+
+  self: ActorSystemBuilder with
+    ServerActorBuilder with
+    BlockChainBuilder with
+    NodeStatusBuilder with
+    PeerManagerActorBuilder with
+    StorageBuilder =>
+
+
+  lazy val fastSyncController = actorSystem.actorOf(
+    FastSyncController.props(
+      peerManager,
+      nodeStatusHolder,
+      blockchain,
+      storagesInstance.storages.mptNodeStorage),
+    "fast-sync-controller")
+
+}
+trait ShutdownHookBuilder {
+
+  def shutdown()
+
+  lazy val shutdownTimeoutDuration = Config.shutdownTimeout
+
+  Runtime.getRuntime.addShutdownHook(new Thread() {
+    override def run(): Unit = {
+      shutdown()
+    }
+  })
+}
+
+trait Node extends NodeKeyBuilder
+  with ActorSystemBuilder
+  with StorageBuilder
+  with BlockChainBuilder
+  with NodeStatusBuilder
+  with PeerManagerActorBuilder
+  with ServerActorBuilder
+  with FastSyncControllerBuilder
+  with JSONRpcServerBuilder
+  with ShutdownHookBuilder
```

Comment:
> With the new Node( you have to build actorSystem, storage, blockChain etc. for each new node type

Not necessarily, I can imagine several ways to avoid repetition (factory methods with default args, directly extending the class, ...)

Not that there's anything inherently wrong with the trait design, but I think for our needs something simpler would suffice. 

That being said, unless I'm missing something, this may simply be a case of personal preference.